lib/core/src/configuration_hook_manager.cpp

```diff
@@ -0,0 +1,44 @@
+#include "irods/configuration_hook_manager.hpp"
+#include "irods/irods_logger.hpp"
+
+#include <unordered_map>
+
+namespace irods::experimental
+{
+    namespace
+    {
+        std::vector<hook> hooks;
+        std::unordered_multimap<std::string, size_t> edge_sets;
+    } // namespace
+
+    void add_hook(const auto& hook_triggers, hook&& hook)
+    {
+        size_t id = hooks.size();
+        // Take ownership, we don't need to deal with any missing closures.
+        // those can be nasty to debug.
+        hooks.emplace_back(std::move(hook));
+        for (const auto& name : hook_triggers) {
+            edge_sets.insert(name, id);
+        }
+    }
+
+    void run_hooks(const nlohmann::json& changes)
+    {
+        // These are hooks that run for every key, seems useful for plugins that
+        // want to be especially nosy about configuration changes
+        auto catch_all_hooks = edge_sets.equal_range("*CATCHALL*");
```

Comment:
Please explain the idea behind `*CATCHALL*`

Reply:
`*CATCHALL*` hooks are triggered on every configuration value change

The reason it looks like that is that I don't think anyone is likely to stumble into a namespace collision with that.